From faadcfae52f4623bfedcdb2e2c53c18b94618cf2 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Sat, 27 Feb 2021 22:19:18 +0000
Subject: [PATCH] fstools: update to git HEAD

Backport generated via the following command including all changes
between `openwrt-21.02` and master branch:

	git log --oneline \
	ffeb37047e85a5efd96890db12710e9d60b4b76a...master  \
	package/system/procd/ | tac | cut -d " " -f 1 | tr '\n' ' '

 bad1835 fstools: add partname volume driver
 19d7d93 libfstools: partname: several fixes
 964d1e3 partname: allow skipping existing 'rootfs_data' partition
 c44b40b overlay: fix syncronizing typo
 b5397a1 fstools: block: fix segfault on mount with no target
 bd7cc8d block: use dynamically allocated target string
 6d8450e blockd: use allocated strings instead of fixed buffers
 d47909e libblkid-tiny: fix buffer overflow
 67d2297 block: match device path instead of assuming /dev/%s
 2aeba88 block: allow autofs and umount commands also on MTD/UBI
 4d4dcfb blockd: detect mountpoint of /dev/mapper/*
 2f42515 block: resolve /dev/mapper/* name for /dev/dm-0 when hotplugging
 39558a1 blockd: also send ubus notification on mount hotplug
 3386b6b blockd: fix trigger name
 cdc9939 blockd: move to its own POSIX process group
 59f7c11 blockd: create mountpoint parent folder if needed
 9cc96af Revert "block: resolve /dev/mapper/* name for /dev/dm-0 when hotplugging"
 06334ac Revert "blockd: detect mountpoint of /dev/mapper/*"
 9ab3551 block: use /dev/dm-* instead of /dev/mapper/*
 5114595 block: allow remove hotplug event to arrive at blockd
 a846c6b blockd: fix length of timeout int passed to ioctl
 1d681ca block: support umount device basename
 46d02c2 block: don't add non-ubifs ubi devices
 cc63933 blockd: send mount.ready when startup has completed
 b7bf185 blockd: make most calls to 'block' asynchronous
 141ac85 libblkid-tiny: fix invalid open syscall return check
 9e26563 libblkid-tiny: install header file to include dir
 d4f0129 blockd: also report target in notifications
 629726d blockd: fix resource leak discovered by coverity scan
 68ae639 libubi: fix several issues discovered by Coverity
 a77c4fa ubi: fix resource leak in legacy codepath
 2e3aca2 block: fix two resources leaks discovered by Coverity
 50e6b20 libfstools: handle open() return value properly in F2FS check
 e1b6811 blockd: include missing libubox/utils.h
 19fd7fc libfstools: make sure file is closed on error
 d390744 libfstools: use uevent instead of relying on custom kernel patch
 77c0288 fstools: fix a couple of minor code problems
---
 package/system/fstools/Makefile          | 8 ++++----
 package/system/fstools/files/blockd.init | 1 +
 package/system/fstools/files/fstab.init  | 6 ++++--
 3 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/package/system/fstools/Makefile b/package/system/fstools/Makefile
index 2da508d541..bc15bfdaf5 100644
--- a/package/system/fstools/Makefile
+++ b/package/system/fstools/Makefile
@@ -8,13 +8,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=fstools
-PKG_RELEASE:=1
+PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/fstools.git
-PKG_MIRROR_HASH:=a485792d90c71cd4fb396ce97f42a57ee4d2a3d78e5f3fd0748270ffb14209e6
-PKG_SOURCE_DATE:=2021-01-04
-PKG_SOURCE_VERSION:=c53b18820756f6f32ad0782d3bf489422b7c4ad3
+PKG_MIRROR_HASH:=e4d8fcdf8f83749553be109c7c39141cc4495ac46d476c8d2dc6a0d8dccfb741
+PKG_SOURCE_DATE:=2021-11-16
+PKG_SOURCE_VERSION:=77c02889177c43df8ebe07bce4d55b7f7358118f
 CMAKE_INSTALL:=1
 
 PKG_LICENSE:=GPL-2.0
diff --git a/package/system/fstools/files/blockd.init b/package/system/fstools/files/blockd.init
index a4ce57d40d..bdd8bbf622 100755
--- a/package/system/fstools/files/blockd.init
+++ b/package/system/fstools/files/blockd.init
@@ -16,6 +16,7 @@ reload_service() {
 start_service() {
 	procd_open_instance
 	procd_set_param command "$PROG"
+	procd_set_param watch block
 	procd_set_param respawn
 	procd_close_instance
 }
diff --git a/package/system/fstools/files/fstab.init b/package/system/fstools/files/fstab.init
index 08d5601bee..03a3993494 100644
--- a/package/system/fstools/files/fstab.init
+++ b/package/system/fstools/files/fstab.init
@@ -1,7 +1,9 @@
 #!/bin/sh /etc/rc.common
-# (C) 2013 openwrt.org
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2013-2020 OpenWrt.org
 
-START=40
+START=11
 
 boot() {
 	/sbin/block mount
-- 
2.35.1

